var/spack/repos/builtin/packages/gromacs/package.py

```diff
@@ -104,13 +98,28 @@ def cmake_args(self):
         else:
             options.append('-DGMX_GPU:BOOL=OFF')
 
-        simd_value = self.spec.variants['simd'].value
-        if simd_value == 'auto':
-            pass
-        elif simd_value == 'none':
+        simd_features = {
+            'sse2': 'SSE2',
+            'sse4_1': 'SSE4.1',
+            'avx128fma': 'AVX_128_FMA',
+            'avx256': 'AVX_256',
+            'axv128': 'AVX2_128',
+            'avx2_256': 'AVX2_256',
+            'avx512': 'AVX_512',
+            'avx512knl': 'AVX_512_KNL',
```

Comment:
`avx512` is kind of a mess, for `knl` the features are 
```
"avx512f",
"avx512pf",
"avx512er",
"avx512cd"
```
I'm not sure what `gromacs` means by just `avx512`, since the degree of support varies across microarchitectures.

Reply:
If the GROMACS build separates the KNL config option out, you could just check for `'target=mic_knl' in spec`

Reply:
knl and skylake avx512 is different.
please look at https://colfaxresearch.com/skl-avx512/ Figure1.

If avx512 and avx512_knl is separated, we can use avx512er.

Reply:
Thanks, fixed!